statement_parser.go

```diff
@@ -399,49 +443,32 @@ func removeCommentsAndTrim(sql string) (string, error) {
 }
 
 // Removes any statement hints at the beginning of the statement.
-func removeStatementHint(sql string) string {
-	parser := &simpleParser{sql: []byte(sql)}
+func (p *statementParser) removeStatementHint(sql string) string {
+	parser := &simpleParser{sql: []byte(sql), statementParser: p}
 	if parser.skipStatementHint() {
 		return sql[parser.pos:]
 	}
 	return sql
 }
 
-type statementsCacheEntry struct {
-	sql    string
-	params []string
-	info   *statementInfo
-}
-
-const statementCacheSize = 1000
-
-// statementsCache contains the results for searching for parameters in SQL strings.
-var statementsCache = func() *lru.Cache {
-	cache, err := lru.New(statementCacheSize)
-	if err != nil {
-		panic(err)
-	}
-	return cache
-}()
-
 // findParams finds all query parameters in the given SQL string.
 // The SQL string may contain comments and statement hints.
-func findParams(sql string) (string, []string, error) {
-	if val, ok := statementsCache.Get(sql); ok {
+func (p *statementParser) findParams(sql string) (string, []string, error) {
+	if val, ok := p.statementsCache.Get(sql); ok {
 		res := val.(*statementsCacheEntry)
 		return res.sql, res.params, nil
 	} else {
-		namedParamsSql, params, err := calculateFindParamsResult(sql)
+		namedParamsSql, params, err := p.calculateFindParamsResult(sql)
 		if err != nil {
 			return "", nil, err
 		}
-		info := detectStatementType(sql)
-		statementsCache.Add(sql, &statementsCacheEntry{sql: namedParamsSql, params: params, info: info})
+		info := p.detectStatementType(sql)
```

Comment:
nit: params here if modified later will modify the cache entry also, if we don't want it we may want to 
Return a defensive copy:
```
paramsCopy := append([]string(nil), res.params...)
return res.sql, paramsCopy, nil
```

Example: https://go.dev/play/p/Pp3dAW8HITu